### PR TITLE
vim-patch:8.1.2229: cannot color number column above/below cursor differently

### PR DESCRIPTION
**Pull request by zeertzjq**
#### vim-patch:8.1.2229: cannot color number column above/below cursor differently

Problem:    Cannot color number column above/below cursor differently.
Solution:   Add LineNrAbove and LineNrBelow. (Shaun Brady, closes vim/vim#624)
https://github.com/vim/vim/commit/efae76ab1a43d5a628d8c2fa4218ace6ba597f5d

### CODE REVIEW

**Comment by zeertzjq** [comment]
Help wanted: How do I check if LineNrAbove and LineNrBelow are not set?

**Comment by zeertzjq** [comment]
> > Help wanted: How do I check if `LineNrAbove` and `LineNrBelow` are not set?
> 
> In what context? If you want to set default highlighting groups you can use `:hi default`. If you want to check in C there is `win_hl_attr`.

I actually wanted to check if these two are set. I tried `win_hl_attr(wp, HLF_LNA) > 0` and `win_hl_attr(wp, HLF_LNB) > 0`, but they seem to be always true.

**Comment by zeertzjq** [comment]
I changed it to `> 1` and the tests seems to pass.

File: src/nvim/screen.c [hunk 1]
```diff
@@ -2832,6 +2832,17 @@ static int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow,
               // when CursorLineNr isn't set?
               char_attr = win_hl_attr(wp, HLF_CLN);
             }
+
+            if (wp->w_p_rnu && lnum < wp->w_cursor.lnum
+                && win_hl_attr(wp, HLF_LNA) > 1) {
```
Comment (bfredl) [inline src/nvim/screen.c:2837]: This a bit strange and doesn't fit our model. I would add "default link LineNrAbove LineNr" in `highlight_init_both` in `syntax.c` and get rid of the check.

**Pull request merged**